src/luarocks/static_flags.lua

```diff
@@ -0,0 +1,89 @@
+
+local static_flags = {}
+package.loaded["luarocks.static_flags"] = static_flags
+
+local util = require("luarocks.util")
+local fetch = require("luarocks.fetch")
+local path = require("luarocks.path")
+local search = require("luarocks.search")
+local fs = require("luarocks.fs")
+
+util.add_run_function(static_flags)
+
+static_flags.help_summary = "Returns all static libraries with flags required for the compiler's Lua flag."
+
+static_flags.help_arguments = "{<name> [<version>]}"
+static_flags.help = [[
+The argument may be the name of locally available module with newest version
+or defined version (optional), which has already built all static libraries.
+
+This command is useful for building an application with static libraries.
+
+Example:
+$ gcc -o myapp main.c -llua $(luarocks static-flags module_name)
+]]
+
+local function print_static_flags(collected_libs)
+   for i = 1, math.floor(#collected_libs / 2) do
+      collected_libs[i], collected_libs[#collected_libs - i + 1] = collected_libs[#collected_libs - i + 1], collected_libs[i]
+   end
+   util.printout(table.concat(collected_libs):sub(1, -2))
+end
+
+local function table_contains(libs_table, name)
+   for _, value in pairs(libs_table) do
+      if value:match("luarocks%-"..name.."%.a") then
+         return true
+      end
+   end
+   return false
+end
+
+local function collect_libs(module_name, flags, libs_table)
+   if not module_name then
+      return nil, "Argument missing. "..util.see_help("static_flags")
+   end
+
+   local name, version = search.pick_installed_rock(module_name:lower(), version, flags["tree"])
+   if not name then
+      util.printout(name..(version and " "..version or "").." is not installed.")
+      return nil, version
+   end
+
+   local rockspec = fetch.load_local_rockspec(path.rockspec_file(name, version), false)
+   local lib_name = "luarocks-"..module_name..".a "
+   if not fs.exists(lib_name) then
+      return nil, "Library ".. lib_name .." is missing. Verify location of library or install with 'luarocks make --static <module>'."
```

Comment:
The trailing space in `lib_name` creates a double space in this error message. See other comments about addressing the trailing space.